### PR TITLE
Prevent compacting NULLCHKs into computed call trees; add new symbol J9JNIMethodIDvTableIndexFieldSymbol

### PR DESCRIPTION
**Pull request by nbhuiyan**
Compacting NULLCHK into computed calls will result in the NULLCHK
not being performed before dispatching into the call target if
there is no need for dereferencing of the reference child to
dispatch.

Furthermore, this PR introduces a new symbol required for eclipse-openj9/openj9#12341

Signed-off-by: Nazim Bhuiyan <nubhuiyan@ibm.com>

### CODE REVIEW

**Comment by nbhuiyan** [comment]
@0xdaryl Requesting review. This is going to be needed for `MethodHandle.linkToVirtual()` transformer

File: compiler/compile/OMRNonHelperSymbols.enum [hunk 1]
```diff
@@ -365,7 +365,12 @@
     */
    j9VMThreadFloatTemp1Symbol,
 
-   OMRlastPrintableCommonNonhelperSymbol = j9VMThreadFloatTemp1Symbol,
+   /** \brief
+    * This symbol represents the vTableIndex field in struct J9JNIMethodID
+    */
+   J9JNIMethodIDvTableIndexFieldSymbol
```
Comment (0xdaryl) [inline compiler/compile/OMRNonHelperSymbols.enum:371]: Is there a comma missing at the end of this line?

Comment (nbhuiyan) [reply compiler/compile/OMRNonHelperSymbols.enum]: Fixed.

File: compiler/optimizer/LocalOpts.cpp [hunk 2]
```diff
@@ -2199,44 +2199,47 @@ bool TR_CompactNullChecks::replacePassThroughIfPossible(TR::Node *currentNode, T
...
+                  (*isTreeTopNode) = true;
+               }
+
+            prevNode->getFirstChild()->recursivelyDecReferenceCount();
+            prevNode->setAndIncChild(0, currentNode);
+            if (child->getOpCodeValue() != TR::loadaddr) //For loadaddr, IsNonNull is always true
+               child->setIsNonNull(false); // it is no longer known that the reference is non null
+
+            if (0 && comp()->useAnchors() &&
+                  currentTree->getNode()->getOpCode().isAnchor())
```
Comment (0xdaryl) [inline compiler/optimizer/LocalOpts.cpp:2230]: Is there value in keeping this code if the whole thing is `if (0 && ...` out?  If not, I'd say delete it.

**Comment by vijaysun-omr** [comment]
I am unsure why this PR does the "Add symbol for J9JNIMethodID.vtableIndex field" commit when it has nothing to do with compacting of null checks.

**Comment by nbhuiyan** [comment]
@vijaysun-omr both these relatively minor changes are required for eclipse-openj9/openj9#12341, and I wanted to combine them into one PR to simplify testing and reviewing. I have updated the PR title.

**Comment by vijaysun-omr** [comment]
Jenkins build all

**Comment by vijaysun-omr** [comment]
jenkins build riscv

**Comment by vijaysun-omr** [comment]
jenkins build riscv

**Comment by vijaysun-omr** [comment]
Checks have passed but I won't be doing another review until it has been approved by @0xdaryl who had commented earlier

**Comment by nbhuiyan** [comment]
@0xdaryl I have addressed your comment and deleted the redundant code.

**Comment by 0xdaryl** [comment]
Jenkins build all

**Review by 0xdaryl — APPROVED** [review]

**Comment by vijaysun-omr** [comment]
Jenkins build aix

**Review by vijaysun-omr — APPROVED** [review]

**Comment by vijaysun-omr** [comment]
Checks have passed now. Merging.

**Pull request merged**